### DISCUSSION TITLE
Custom truststore

### DISCUSSION THREAD

**Discussion by aseigi** [strimzi/strimzi-kafka-bridge#783] [Q&A]
Hello, I'm trying to set a custom JKS truststore (kafka.ssl.truststore.location, kafka.ssl.truststore.password and kafka.ssl.truststore.type) in my bridge...but checking this piece of code* I think its not possible. Can anyone help me?

*https://github.com/strimzi/strimzi-kafka-bridge/blob/main/bin/docker/kafka_bridge_config_generator.sh

**Comment by scholzj** [comment]
That script is normally used when deploying the Bridge from the Strimzi operator. If you want to run and configure it yourself, you can just put these options into the config file (check the example from `contig/`) ans start the bridge with it.

**Reply by aseigi** [reply]
I've deployed using operator but I'm gonna check the alternative. Thanks for the reply.

**Reply by scholzj** [reply]
If you deploy it through the operator, then you cannot use custom TLS configuration like this. You would need to use the `.spec.authentication` section in the `KafkaBridge` custom resource.

**Reply by aseigi** [reply]
Yeah, I've set up a authentication section but I also need a custom truststore....Maybe the only solution is either customize it and run by itself or build a new component for that......

**Reply by scholzj** [reply]
Well, you can simply extract the public and private keys from the JKS truststore, add it to a secret and just use it. I think google will find you some exact guide - it should be just a few commands.
